hivelvet-backend/tests/src/Actions/Users/AddTest.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * Hivelvet open source platform - https://riadvice.tn/
+ *
+ * Copyright (c) 2022 RIADVICE SUARL and by respective authors (see below).
+ *
+ * This program is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU Lesser General Public License as published by the Free Software
+ * Foundation; either version 3.0 of the License, or (at your option) any later
+ * version.
+ *
+ * Hivelvet is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+ * PARTICULAR PURPOSE. See the GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along
+ * with Hivelvet; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace Actions\Users;
+
+use Enum\UserRole;
+use Fake\UserFaker;
+use ReflectionException;
+use Test\Scenario;
+
+/**
+ * @internal
+ * @coversNothing
+ */
+final class AddTest extends Scenario
+{
+    final protected const ADD_USER_ROUTE = 'POST /users/add';
+    protected $group                     = 'Action User Add';
+
+    /**
+     * @param $f3
+     *
+     * @throws ReflectionException
+     *
+     * @return array
+     */
+    public function testExistingUser($f3)
+    {
+        $test = $this->newTest();
+
+        $user = UserFaker::create(UserRole::LECTURER);
+        $data = ['data' => ['username' => $user->username, 'email' => $user->email, 'password' => UserRole::LECTURER,  'role' => 2]];
```

Comment:
We have a convention where ADMIN role id is 1 and LECTURER role id 2. Both should go to static const, as example `UserRole::ADMINISTRATOR_ID` equals to `1`.